PubSub/tests/System/ManageTopicsTest.php

```diff
@@ -77,4 +77,44 @@ public function testReloadTopic($client)
         $this->assertTrue($client->topic($shortName)->exists());
         $this->assertEquals($topic->name(), $topic->reload()['name']);
     }
+
+    /**
+     * @dataProvider clientProvider
+     */
+    public function testUpdateTopic($client)
+    {
+        $topics = $client->topics();
+        $topic = $topics->current();
+
+        $policy = [
+                'allowedPersistenceRegions' => ['us-central1', 'us-east1']
+        ];
+
+        $topic->update([
+            'messageStoragePolicy' => $policy
+        ]);
+
+        $this->assertEquals($policy, $topic->info()['messageStoragePolicy']);
+    }
+
+    /**
+     * @dataProvider clientProvider
+     */
+    public function testUpdateTopicWithUpdateMask($client)
+    {
+        $topics = $client->topics();
+        $topic = $topics->current();
+
+        $labels = [
+            'foo' => 'bar'
+        ];
+
+        $topic->update([
+            'labels' => $labels
+        ], [
+            'updateMask' => [ 'labels' ]
+        ]);
+
+        $this->assertEquals($labels, $topic->info()['labels']);
```

Comment:
Could we also have a test ensuring an update on a specific label works as expected? (vs modifying them all)

Reply:
I believe that labels must be updated as a set. You can't target specific labels in an update mask (e.g. `labels.0` or `labels[0]`).

Reply:
I believe that labels must be updated as a set. You can't target specific labels in an update mask (e.g. labels.0 or labels[0]).

Reply:
That makes sense, too bad!